src/core/org/apache/jmeter/gui/TreeState.java

```diff
@@ -0,0 +1,84 @@
+package org.apache.jmeter.gui;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.swing.JTree;
+
+public interface TreeState {
+
+    /**
+     * Restore tree expanded and selected state
+     *
+     * @param guiInstance GuiPackage to be used
+     */
+    void restore(GuiPackage guiInstance);
+
+    final static TreeState NOTHING = new TreeState() {
```

Comment:
`static final` to conform with java conventions and the anonymous class implementing `NOTHING` could be rewritten as a lambda function.